### DISCUSSION TITLE
Best current way to import internal jars into a tycho project?

### DISCUSSION THREAD

**Discussion by hammyau** [eclipse-tycho/tycho#449] [Q&A] [answered]
I have a couple of jars that are based around ANTLR parsers and I currently just place them into a lib directory and my Eclipse IDE build is happy.

Tycho is not as happy :-(

I have read a few bits and pieces about different techniques to get a jar into your RCP build.
Some of which seem to be pretty old.

Creating it as p2 entry via its own pom? A pom first build and then a manifest first build. Not sure I get this?

There is a mechanism of adding it to the Target Platform and it is then wrapped. I can see the wrapped version but it does not seem to be used. Compile phase does not find the code. And so get compilation errors.

Convert my parser project to a plugin project and add to the tycho build? Still then need the stage to generate the ANTLR files.

What is the current and recommended way of supporting external jars? Can anyone enlighten me?

Thanks in advance.

**Comment by laeubi** [comment, marked as answer]
Just convert those to plugin projects and bind the ANTLR generation to the `generate-sources` phase should work without an issue.

Beside that you should describe how the libs are currently used inside eclipse, again here sharing a simple example would make it easier to decide whats possible.

**Reply by hammyau** [reply]
Hi Christoph, the libs are just worker classes. There is a function to pass in the text to be processed and then providing there are no errors a getter to return some derived information.

**Reply by laeubi** [reply]
Sorry if the question is misleading. I mean you wrote 'just place them into a lib directory and my Eclipse IDE build is happy' I assume there is no global folder where you drop those and Eclipse does not take them magically into account isn't it? So there should be some specific path (inside the project? configured as a user lib?) and some project configuration involved (added to the bundle-classpath?).

**Reply by hammyau** [reply]
Well, I didn't convert the jars to plugins in the end I just left then to be wrapped which seems to work.
I will however look at doing that in the future since I think that would be a better solution.

So will close this as answered along with #418
